chat_ui.py

```
@@ -214,60 +214,312 @@ def _build() -> gr.Blocks:
+                                    label="Provider", choices=[], interactive=True,
+                                )
+                                switch_status = gr.Markdown("")
+
+                        def load_model():
+                            return settings["get_model_info"]()
+
+                        app.load(fn=load_model, outputs=[model_display])
+                        refresh_model_btn.click(fn=load_model, outputs=[model_display])
+
+                        if provider_dropdown is not None:
+                            def load_provider_choices():
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `load_provider_choices`

(ANN202)